src/components/Calculator.js

```diff
@@ -122,17 +104,24 @@ class Calculator extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
-      result: 0,
+      total: null,
+      next: null,
     };
+
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(e) {
+    this.setState((state) => (calculate(state, e.target.innerText)));
   }
 
   render() {
-    const { result } = this.state;
+    const { total, next } = this.state;
     return (
       <div className="Calculator">
-        <section className="Result">{result}</section>
+        <section className="Result">{total === null ? next : total}</section>
         <section className="Pad">
-          {buttons.map((button) => <button key={button.id} type="button" className={`Button ${button.id} ${button.color}`}>{button.symbol}</button>)}
+          {buttons.map((button) => (<button key={button.id} onClick={this.handleClick} type="button" className={`Button ${button.id} ${button.color}`}>{button.symbol}</button>))}
```

Comment:
- [OPTIONAL] Your calculator is working fine. However, I will suggest you update your UI by adding a display for the select operator ( +, - , %,  x, /)
As of now, only the selected numbers and the result are shown.

Please take a look at the test gif:|
:--------------------------------------------------:|
![ma](https://user-images.githubusercontent.com/39523312/158877672-cd599b51-3688-4680-826d-d624ea00dc38.gif)

Reply:
Thanks, @mmsesay, I am going to implement this now!
You just gave me an idea :tada: